website/pages/docs/concepts/password-policies.mdx

```diff
@@ -0,0 +1,290 @@
+
+### Preventing Bias
+When using the [modulo operation](https://en.wikipedia.org/wiki/Modulo_operation) to generate a password,
+you must be very careful to prevent the introduction of bias. When generating a random number between
+[0-255] for a charset that has a length that isn't evenly divisible into 256, some of the first characters
+in the charset may be selected more frequently than the remaining characters.
+
+To demonstrate this, let's simplify the math. Assume that we have a charset of length 10: `abcdefghij`.
+Let's also assume that our RNG generates values `[0-25]`. The first 10 values `0-9` correspond to each
+character in our charset. The next 10 values `10-19` also correspond to each character in our charset.
+However, the next 6 values `20-25` correspond to only the first 6 characters in the charset. This means
+that those 6 characters `abcdef` can be selected more often than the last 4 characters `ghij`.
+
+In order to prevent this from happening, we calculate the maximum value that we can allow an index to be.
+This is based on the length of the charset we are selecting from. In the example above, the maximum index
+value we should allow is 19 as that represents the largest integer multiple of the length of the charset
+array that is less than the maximum value that our RNG can generate. When our RNG generates any values
+larger than our maximum allowed value, that number is ignored and we continue to the next number. Passwords
+do not lose any length because we continue generating numbers until the password is fully filled in to the
+length requested.
+
+## Password Policy Syntax
+Password policies are defined in [HCL](https://github.com/hashicorp/hcl) or JSON which defines
+the length of the password and a set of rules a password must adhere to.
+
+Here is a very simple policy which generates 20 character passwords from lowercase characters:
+
+```hcl
+length = 20
+rule "charset" {
+  charset = "abcdefghijklmnopqrstuvwxyz"
+}
+```
+
+Multiple rules may be specified, including multiple rules of the same type. For instance, the following
+policy will generate a 20 character password with at least one lowercase letter, at least one uppercase
+letter, at least one number, and at least one symbol from the set `!@#$%^&*`:
+
+```hcl
+length = 20
+rule "charset" {
+  charset = "abcdefghijklmnopqrstuvwxyz"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "0123456789"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "!@#$%^&*"
+  min-chars = 1
+}
+```
+
+At least one charset must be specified for a policy to be valid. In order to generate a password, a charset
+must be available to select characters from and password policies do not have a default charset.
+The following policy is **NOT** valid and will be rejected:
+
+```hcl
+length = 20
+```
+
+## Performance Characteristics
```

Comment:
I wonder if this section would fit better at the end of this page, since **Available Rules** is closely related to **Password Policy Syntax**.